week2/index.js

```diff
@@ -0,0 +1,135 @@
+
+const fs = require('fs');
+const readline = require('readline');
+let todos = [];
+
+
+if (process.argv[2] !== undefined) {
+    start();
+} else {
+    console.log('Please specify an option to use, here is the manual\n');
+    manualReader('./help.txt');
+}
+
+
+function start() {
+    switch (process.argv[2].trim().toLowerCase()) {
+        case 'help':
+            manualReader('./help.txt');
+            break;
+        case 'add':
+            checkFileContent().then((item) => addTodoItem(item)).catch((err) => console.log(err));
+            break;
+        case 'remove':
+            itemRemover(process.argv[3]);
+            break;
+        case 'list':
+            console.log('The items on your todo list: ');
+            todoFileReader('./todos.txt');
+            break;
+        case 'reset':
+            itemsResetter();
+            break;
+        default:
+            console.log('No such option! Take a look at the manual\n');
+            manualReader('./help.txt');
+            break;
+    }
+}
+
+
+function manualReader(fileName) {
+    fs.readFile(fileName, 'UTF-8', (err, data) => {
+        console.log(data);
+    });
+}
+
+//  display the todo file once called
+function todoFileReader(fileName) {
+    fs.exists(fileName, (exists) => {
+        //see if file exists to add the existing items to the array
+        if (exists) {
+            fs.readFile(fileName, 'UTF-8', (err, data) => {
+                if (data === "") {
+                    console.log('There are no items yet, try using \'node index.js add "some todo item"\'');
+                    return;
+                }
+                data = data.split('\r\n');
+                let text = data.reduce(((last, item, index) => last += `${index + 1}. ${item}\n`), ``);
+                console.log(text);
+            });
+        } else {
+            console.log('There are no items yet, try using \'node index.js add "some todo item"\'');
+        }
+    });
+}
+
+
+
+//check if there's already a file and to push the item in
+
+function checkFileContent() {
+    return new Promise(function (resolve, reject) {
+        fs.exists('./todos.txt', (exists) => {
+            if (!exists) {
+                resolve(process.argv[3]);
+                return;
+            }
+            fs.readFile('todos.txt', 'UTF-8', (err, data) => {
+                todos = data.trim().split('\r\n');
```

Comment:
Your use of global variables is a bit awkward here. You generally don't want functions to manipulate or change variables that exist outside their scope (in a higher scope). It'd be better to resolve the promise with the parsed data.

Reply:
i dont knw why i still dont get the promiss consept as wll thts why i have alot of mistakes in it -__-